gcode/src/lexer.rs

```diff
@@ -175,6 +177,23 @@ impl<'input> Lexer<'input> {
             },
         })
     }
+    
+    fn tokenize_newline(&mut self) -> Option<Token<'input>> {
+        let start = self.current_position;
+        let line = self.current_line;
+        let value = "\n";
+        self.current_position += 1;
+        self.current_line += 1;
+        Some(Token {
+            kind: TokenType::Newline,
+            value,
+            span: Span {
+                start,
+                line,
+                end: start + 1,
+            },
+        })
+    }
```

Comment:
If the only way this function can be called is when we're looking at a `\n` character, would it be enough to add something like `debug_assert!(self.rest().starts_with("\n"))` and return a `Token<'input>`?

Otherwise, we should actually check for the newline character and return `None` if it isn't found.

Either way, I think we should try to use the newline characters from the original stream instead of writing `let value = "\n"`.

Reply:
I think what you're getting at is "this code has a weird smell", and I think you're right. 😁  But I'm honestly not sure how to construct this better. If we don't explicitly set `value = "\n"`, we'd have to `chomp()` or somehow otherwise pull the newline from the string, which is a computational expense that seems superfluous.

As for your suggestion
> would it be enough to add something like `debug_assert!(self.rest().starts_with("\n"))` and return a `Token<'input>`

I don't think I understand. The `debug_assert` is something I could add; seems reasonable. But how would you return the `Token<'input>`?